File: app/src/main/kotlin/io/mikupush/ui/Theme.kt
package io.mikupush.ui

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import org.slf4j.LoggerFactory

private val DarkColorScheme = darkColorScheme(
    primary = Verdigris,
    secondary = RichBlack,
    background = Dark,
    surface = Dark,
    onPrimary = Color.Black,
    onSecondary = Color.White,
    onBackground = Color.White,
    onSurface = Color.White,
    primaryContainer = Verdigris,
    onPrimaryContainer = Color.Black,
    outline = Verdigris,
    secondaryContainer = Verdigris,
    onSecondaryContainer = Color.Black,
)

private val LightColorScheme = lightColorScheme(
    primary = CaribbeanCurrent,
    secondary = Night,
    background = Color.White,
    surface = Color.White,
    onPrimary = Color.White,
    onSecondary = Color.White,
    onBackground = Color.Black,
    onSurface = Color.Black,
    primaryContainer = CaribbeanCurrent,
    onPrimaryContainer = Color.White,
    secondaryContainer = TiffanyBlue,
    onSecondaryContainer = Color.Black,
    surfaceContainerHighest = AntiFlashWhite
)

private val logger = LoggerFactory.getLogger("Theme")

@Composable
fun MikuPushTheme(content: @Composable () -> Unit) {
    val colorScheme = if (isSystemInDarkTheme()) {
        logger.debug("use dark theme")
        DarkColorScheme
    } else {
        logger.debug("use light theme")
        LightColorScheme
    }

    MaterialTheme(
        colorScheme = colorScheme,
        content = content
    )
}

File: lib/theme.dart
import 'dart:ui';

import 'package:flutter/material.dart';

class AppColors {
  static const caribbeanCurrent = Color(0xFF057073);
  static const tiffanyBlue = Color(0xFF9AD5D5);
  static const night = Color(0xFF0A0C0D);
  static const antiFlashWhite = Color(0xFFEAEAEB);
  static const verdigris = Color(0xFF38B0B6);
  static const richBlack = Color(0xFF020516);
  static const keppel = Color(0xFF5ABAAC);
  static const dark = Color(0xFF1C1B1F);
  static const light = Color(0xFFE6E1E5);
}

final lightThemeData = ThemeData(
  colorScheme: ColorScheme.light(
      surface: Colors.white,
      onSurface: Colors.black,
      primary: AppColors.caribbeanCurrent,
      onPrimary: Colors.white,
      secondary: AppColors.night,
      onSecondary: Colors.white,
      primaryContainer: AppColors.caribbeanCurrent,
      onPrimaryContainer: Colors.white,
      secondaryContainer: AppColors.tiffanyBlue,
      onSecondaryContainer: Colors.black,
      surfaceContainerHighest: AppColors.antiFlashWhite
  ),
);

final darkThemeData = ThemeData(
  colorScheme: ColorScheme.dark(
      surface: AppColors.dark,
      onSurface: Colors.white,
      primary: AppColors.verdigris,
      onPrimary: Colors.black,
      secondary: AppColors.richBlack,
      onSecondary: Colors.white,
      primaryContainer: AppColors.verdigris,
      onPrimaryContainer: Colors.black,
      secondaryContainer: AppColors.verdigris,
      onSecondaryContainer: Colors.black,
      surfaceContainerHighest: Colors.black
  ),
);
